platform/linux-generic/pktio/loop.c

```diff
@@ -218,6 +218,11 @@ static int loopback_recv(pktio_entry_t *pktio_entry, int index ODP_UNUSED,
 
 				ret = _odp_cls_classify_packet(pktio_entry, pkt_addr,
 							       &new_pool, pkt_hdr);
+				if (ret < 0)
+					odp_atomic_inc_u64(
+						&pktio_entry->s.stats_extra
+							 .in_discards);
+
```

Comment:
Looks like this branch is identical for all pktio devices except for TAP, which currently doesn't support statistics anyway. So incrementing the counter could potentially be moved inside `_odp_cls_classify_packet ()`, which could also mitigate some of the performance overhead I'm measuring with the current PR.

Reply:
Technically, it could. But, again, the idea is that classify_packet() only classifies a packet, and doesn't do anything else. Pktio statistics are not the responsibility of the classifier.

Reply:
Makes sense. I'm working on a `pkt_dpdk_t` refactoring PR, which will mitigate the performance degradation.